elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java

```diff
@@ -536,6 +565,58 @@ private static <T> Predicate<T> hasMember(Path fieldPath, List<Object> values, R
         };
     }
 
+    private static <T> Predicate<T> subsetOf(Path fieldPath, List<Object> values, RequestScope requestScope) {
+        return (T entity) -> {
+            Type<?> valueClass = fieldPath.lastElement().get().getFieldType();
+
+            Object leftHandSide = getFieldValue(entity, fieldPath, requestScope);
+
+            BiPredicate<Object, Object> predicate = (a, b) -> a.equals(b);
+
+            List<?> rightHandSide = values.stream().map(value -> CoerceUtil.coerce(value, valueClass)).toList();
+
+            if (leftHandSide instanceof Collection<?> collection && !valueClass.isAssignableFrom(COLLECTION_TYPE)) {
```

Comment:
How is this if condition possible?

Reply:
https://github.com/yahoo/elide/blob/6e2294ecb3ff076d9d9b831a6be8ac092cd12412/elide-core/src/main/java/com/yahoo/elide/core/filter/predicates/SubsetOfPredicate.java#L40-L42

For instance the inputs could be 

```java
new SubsetOfPredicate(authorAwardsElement, "Bookery prize", "Test");
```

In which case the `leftHandSide` is a collection and the value class is `String` (which isn't a `Collection`).

Reply:
The leftHandSide of a predicate author.awards should be a list - not a string.  I imagine Elide would not make a pathElement like the one above based on the model author.awards.

Reply:
Sorry, I think my example in the explanation wasn't correct.

I think this seems to be related to how the `getFieldValue` method works with respect to a to many relationship.

The integration tests that cover these conditions is in `com.yahoo.elide.tests.FilterIT` in the to many relationship tests, for instance `testRootSubsetOfToManyRelationship`.

The logic is also similar to the existing logic at

https://github.com/yahoo/elide/blob/68ddf67f2d793bd64f195fd63d6fcf045a98aea4/elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java#L601-L617

Reply:
You are correct.  When traversing a toMany relationship, it will explode the leftHandSide into a list, even if the last element of the predicate is a scalar type.